src/ngraph/frontend/fluid/operators/matmul.cpp

```diff
@@ -0,0 +1,115 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "ngraph/frontend/fluid/operators/matmul.hpp"
+#include <memory>
+#include <numeric>
+#include "ngraph/builder/matmul_factory.hpp"
+#include "ngraph/builder/reshape.hpp"
+#include "ngraph/op/reshape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo fluid::MatMul::type_info;
+fluid::MatMul::MatMul(const Output<Node>& A,
+                      const Output<Node>& B,
+                      const bool& transpose_a,
+                      const bool& transpose_b)
+    : FusedOp(OutputVector{A, B})
+    , m_transpose_a{transpose_a}
+    , m_transpose_b{transpose_b}
+{
+    constructor_validate_and_infer_types();
+}
+template <class Input>
+void DecomposeLogic(Input& input, bool transpose, bool reverse = false)
+{
+    auto rank = input.get_shape().size();
+    if (rank < 2)
+    {
+        if (rank)
+        {
+            if (reverse)
+            {
+                input =
+                    make_shared<op::Reshape>(input, AxisVector{0}, Shape{input.get_shape()[0], 1});
+            }
+            else
+            {
+                input =
+                    make_shared<op::Reshape>(input, AxisVector{0}, Shape{1, input.get_shape()[0]});
+            }
+        }
+        else
+        {
+            input = make_shared<op::Reshape>(input, AxisVector{}, Shape{1, 1});
+        }
+        rank = 2;
+    }
+    if (transpose)
+    {
+        vector<size_t> axes_order(rank);
+        iota(axes_order.begin(), axes_order.end(), 0);
+        swap(axes_order[rank - 1], axes_order[rank - 2]);
+        input = builder::reorder_axes(input, axes_order);
+    }
+}
+inline NodeVector remove_1(std::shared_ptr<ngraph::Node> input_node)
+{
+    auto input_shape = input_node->get_shape();
+    AxisVector axis(input_shape.size());
+    iota(axis.begin(), axis.end(), 0);
+    Shape shape(input_shape.begin(), input_shape.end());
+    auto b_remove = std::remove(shape.begin(), shape.end(), 1);
+    shape.erase(b_remove, shape.end());
+    Output<Node> node(input_node);
+    auto reshape = make_shared<op::Reshape>(node, axis, shape);
+    NodeVector final_vector{reshape};
+    return final_vector;
+}
+void fluid::MatMul::pre_validate_and_infer_types()
+{
+    element::Type input_element_type = get_input_element_type(0);
+    NODE_VALIDATION_CHECK(this,
+                          input_element_type.is_dynamic() || input_element_type.is_real(),
+                          "Argument element type must be f16, bf16, f32, f64 or dynamic (got ",
+                          input_element_type,
+                          ").");
+    set_output_type(0, get_input_element_type(0), PartialShape::dynamic());
```

Comment:
We still want this to work with static shape version, so please set it as dynamic only when it is dynamic.